animations-dbus/animations-dbus-server-surface.c

```diff
@@ -0,0 +1,691 @@
+                          &animation_effect_id,
+                          &local_error))
+    {
+      g_dbus_method_invocation_return_gerror (invocation,
+                                              g_steal_pointer (&local_error));
+      return TRUE;
+    }
+
+  AnimationsDbusServerEffect *server_animation_effect =
+    animations_dbus_server_lookup_animation_effect_by_ids (priv->server,
+                                                           animation_manager_id,
+                                                           animation_effect_id,
+                                                           &local_error);
+
+  /* Insert the animation effect into the table */
+  if (server_animation_effect == NULL)
+    {
+      g_dbus_method_invocation_return_gerror (invocation,
+                                              g_steal_pointer (&local_error));
+      return TRUE;
+    }
+
+  /* Newly attached effects take priority over old ones */
+  if (!animations_dbus_server_surface_attach_effect_with_queue_func (server_surface,
+                                                                     event,
+                                                                     server_animation_effect,
+                                                                     g_queue_push_head,
+                                                                     &local_error))
+    {
+      g_dbus_method_invocation_return_gerror (invocation,
+                                              g_steal_pointer (&local_error));
+    }
+
+  animations_dbus_animatable_surface_complete_attach_animation_effect (animatable_surface,
+                                                                       invocation);
+  return TRUE;
+}
+
+static gboolean
+animations_dbus_server_surface_detach_animation_effect (AnimationsDbusAnimatableSurface *animatable_surface,
+                                                        GDBusMethodInvocation           *invocation,
+                                                        const char                      *event,
+                                                        const char                      *effect_path)
+{
+  AnimationsDbusServerSurface *server_surface = ANIMATIONS_DBUS_SERVER_SURFACE (animatable_surface);
+  AnimationsDbusServerSurfacePrivate *priv = animations_dbus_server_surface_get_instance_private (server_surface);
+  unsigned int animation_manager_id = 0;
+  unsigned int animation_effect_id = 0;
+  GQueue *attached_effects_for_events = NULL;
+  g_autoptr(GError) local_error = NULL;
+
+  /* Validate that the passed in effect_path is a valid object path */
+  if (!parse_effect_path (effect_path,
+                          &animation_manager_id,
+                          &animation_effect_id,
+                          &local_error))
+    {
+      g_dbus_method_invocation_return_gerror (invocation,
+                                              g_steal_pointer (&local_error));
+      return TRUE;
```

Comment:
Should it return `FALSE`?

Reply:
AIUI, `TRUE` here means that the method call handler ran (regardless the error state is in the invocation return value). There's another place below where I return `FALSE` in error though, I'll fix that.